### PR TITLE
Replace hammerjs with ember-gesture-modifiers, fixes #138

### LINKED ISSUE #138: Replace hammerjs (pan support) with something lighter
https://github.com/nickschot/ember-gesture-modifiers

Panning is used here https://github.com/knownasilya/ember-toggle/blob/master/addon/components/x-toggle-switch/template.hbs#L8 and the tests seem to have stopped working: https://github.com/knownasilya/ember-toggle/blob/master/tests/integration/components/x-toggle/component-test.js#L287 (not firing)

### PR DESCRIPTION
**Pull request by lolmaus**
I've tried to tackle #138 and `ember-gesture-modifiers` don't work for me: the action does not fire. 

I tried debugging the modifier a bit and discovered that the [addEventListeners](https://github.com/nickschot/ember-gesture-modifiers/blob/master/addon/modifiers/did-pan.js#L14) hook does fire but actions below do not.

@nickschot, can you please have a look and see what I'm missing? 🙏

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -50,6 +49,7 @@
     "ember-cli-terser": "^4.0.0",
     "ember-disable-prototype-extensions": "^1.1.3",
     "ember-export-application-global": "^2.0.1",
+    "ember-gesture-modifiers": "^0.2.0",
```
Comment (knownasilya) [inline package.json:52]: This would have to move to dependencies

**Comment by nickschot** [comment]
Just checked locally what is happening and everything seems to be working as intended except that it is mirrored (drag left, toggle right). 

One not of importance here is that currently ember-gesture-modifiers only supports touch events, not pointer events, so it will only work on touch screens right now. I have started working on pointer event support in ember-gesture-modifiers, but did not finish it yet.

If we want to keep ember-toggle exactly like it is we'll need the pointer event support first before merging it.

**Comment by lolmaus** [comment]
Ah! This makes sense. 😬

I'm not currently using `ember-toggle` in any of my active projects, but if I did, I would prefer the mouse drag to keep working. That would definitely be a breaking change otherwise. Breaking the user rather than CI. 😅 

@nickschot Ping in in the PR when you have it for more 👀.

**Comment by nickschot** [comment]
I've made a branch with PointerEvents support which _should_ make this branch work. https://github.com/nickschot/ember-gesture-modifiers/pull/16 Would be great if we could try the branch before I do a release.

**Comment by knownasilya** [comment]
Tests are running. Thanks for working on this @nickschot

**Comment by knownasilya** [comment]
@nickschot do the way we test it need to be updated?

**Comment by nickschot** [comment]
Oh yes! You probably used something like my pan helper? You'll need to let it create a PointerEvent now instead of a TouchEvent.

**Comment by knownasilya** [comment]
Yeah those events were hammerjs specific I believe.

**Comment by knownasilya** [comment]
@nickschot could you export the pan helper as part of `addon-test-support`?

**Comment by nickschot** [comment]
> @nickschot could you export the pan helper as part of `addon-test-support`?

Done! 

```
import { pan } from 'ember-gesture-modifiers/test-support';

...

await pan('.css-selector', 'right');
```

**Comment by knownasilya** [comment]
@nickschot waiting on tests, but I think you can merge that pointer events PR.

**Comment by nickschot** [comment]
@knownasilya great! I'll prepare a release then.

**Comment by nickschot** [comment]
Done: https://github.com/nickschot/ember-gesture-modifiers/releases/tag/v1.0.0

**Comment by knownasilya** [comment]
What do you guys think about semver for this change. It seems like a patch, but people could be using hammerjs from this addon. Thoughts

**Pull request merged**